game-core/src/main/java/games/strategy/triplea/delegate/EndTurnDelegate.java

```diff
@@ -206,51 +195,73 @@ private String addUnitCreatedResources(final IDelegateBridge bridge) {
     return endTurnReport.toString();
   }
 
+  /**
+   * Find all of the resources that will be created by units on the map.
+   */
+  public static IntegerMap<Resource> findUnitCreatedResources(final PlayerID player, final GameData data) {
```

Comment:
Just thinking ahead...  Now that all these utility methods are being made public on `EndTurnDelegate`, that may suggest this class is no longer the right place to host them.  It could be my lack of experience with the engine, but I would never think to look for these types of utilities in this class. :smile:

Reply:
Yeah, generally I'd like to move lots of the logic outside of delegates as most of it the AI can take advantage of. Delegates should really be focused primarily just on the Changes to game data.

Reply:
Yeah, what caught my attention is that a superclass (`AbstractEndTurnDelegate`) is now calling static methods on a subclass (`EndTurnDelegate`).  Generally, not a good smell. :smile:

Reply:
Yeah, I'll take a further look at if it makes sense to move some of these methods around a bit in follow on PRs.